tests/integration/servicecatalog/service_test.go

```diff
@@ -36,6 +36,40 @@ var _ = Describe("odoServiceE2e", func() {
 		})
 	})
 
+	Context("create service with Env non-interactively", func() {
+		JustBeforeEach(func() {
+			project = helper.CreateRandProject()
+			context = helper.CreateNewContext()
+			app = helper.RandString(7)
+			originalDir = helper.Getwd()
+			helper.Chdir(context)
+		})
+		JustAfterEach(func() {
+			helper.DeleteProject(project)
+			helper.DeleteDir(context)
+			helper.Chdir(originalDir)
+		})
+
+		It("should be able to create postgresql with env", func() {
+			oc.ImportJavaIsToNspace(project)
+			helper.CopyExample(filepath.Join("source", "openjdk-sb-postgresql"), context)
+
+			// Local config needs to be present in order to create service https://github.com/openshift/odo/issues/1602
+			helper.CmdShouldPass("odo", "create", "java", "sb-app", "--project", project)
+
+			helper.CmdShouldPass("odo", "service", "create", "dh-postgresql-apb", "--project", project, "--plan", "dev",
+				"-p", "postgresql_user=lukecage", "-p", "postgresql_password=secret",
+				"-p", "postgresql_database=my_data", "-p", "postgresql_version=9.6")
```

Comment:
Modify this to use `--app` and `--project` if you remove the earlier lines that are creating a context first.

Reply:
resolved